lib/util/vsac_api.rb

```
@@ -0,0 +1,245 @@
+
+        # if username and password were provided use them to get a ticket granting ticket
+        elsif !options[:username].nil? && !options[:password].nil?
+          @ticket_granting_ticket = get_ticket_granting_ticket(options[:username], options[:password])
+        end
+      end
+
+      ##
+      # Gets the list of profiles. This may be used without credentials.
+      #
+      # Returns a list of profile names.
+      def get_profiles
+        profiles_response = RestClient.get("#{@config[:utility_url]}/profiles")
+        profiles = []
+
+        # parse xml response and get text content of each profile element
+        doc = Nokogiri::XML(profiles_response)
+        profile_list = doc.at_xpath("/ProfileList")
+        profile_list.xpath("//profile").each do |profile|
+          profiles << profile.text
+        end
+
+        return profiles
+      end
+
+      ##
+      # Gets the list of programs. This may be used without credentials.
+      #
+      # Returns a list of program names.
+      def get_programs
+        programs_response = RestClient.get("#{@config[:utility_url]}/programs")
+        program_names = []
+
+        # parse json response and return the names of the programs
+        programs_info = JSON.parse(programs_response)['Program']
+        programs_info.each do |program|
+          program_names << program['name']
+        end
+
+        return program_names
+      end
+
+      ##
+      # Gets the details for a program. This may be used without credentials.
+      #
+      # Returns the JSON parsed response for program details.
+      def get_program_details(program = nil)
+        # if no program was provided use the one in the config or default in constant
+        if program == nil
+          program = @config.fetch(:program, DEFAULT_PROGRAM)
+        end
+
+        # parse json response and return it
+        return JSON.parse(RestClient.get("#{@config[:utility_url]}/program/#{URI.escape(program)}"))
```

If the `program` that gets passed in is invalid, what error response would this return? Should the api read the response and throw an error if it is invalid? This question also applies to the `get_releases_for_program` function below.

---

Good catch. I'll make it handle that.